modules/post/multi/gather/grub_password.rb

```diff
@@ -0,0 +1,61 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+  include Msf::Post::Unix
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'         => '*nix Gather Grub Password',
+      'Description'  => %q{
+        This module gathers grub passwords from grub bootloader config file.
+      },
+      'License'      => MSF_LICENSE,
+      'Author'       =>
+        [
+          'Garvit Dewan <d.garvit[at]gmail.com>' # @dgarvit
+        ],
+      'Platform'     => ['linux', 'osx', 'unix', 'solaris', 'bsd'],
+      'SessionTypes' => ['meterpreter', 'shell'],
+      'References'   => [
+        ['URL', 'https://help.ubuntu.com/community/Grub2/Passwords#Password_Encryption']
+      ]
+    ))
+  end
+
+  def run
+    targets = [
```

Comment:
Did you find any additional default paths that may be used by GRUB during your research?

Reply:
Yes, https://opensource.com/article/17/3/introduction-grub2-configuration-linux mentions `/boot/grub2/grub.cfg`. Although the directory does not exist on my machine (Ubuntu 18.04). So, I would like to know if I should add it to the targets or not.

Reply:
Include any path that you can validate based on multiple data points. Even better if you can verify yourself.

In this instance, `/boot/grub2/grub.cfg` is a valid path for GRUB.

Reply:
Okay!

Reply:
`/boot/grub2/user.cfg`